### PR TITLE
[Builder] Clone GitHub repository to `/home/mlrun_code` instead of `/tmp`

### PR DESCRIPTION
**Pull request by gtopper**
Second attempt at [ML-4741](https://jira.iguazeng.com/browse/ML-4741) after reverting #4434.

### CODE REVIEW

**Review by alonmr — COMMENTED** [review]
please take into consideration the original reason this was moved to tmp.
It was due to permissions issues with spark image.
you can also use clone_target_dir (maybe have a dedicated dir in the spark image)
you can't assume anything on the running user due to security context.

**Comment by gtopper** [comment]
@theSaarco, I updated with the PR as per your suggestion. I ran the builder tests, the `test_run_git_build` system test, and the relevant [ML-4741](https://jira.iguazeng.com/browse/ML-4741) notebook.

@alonmr, are you good with it too? You guys commented at about the same time so I assume you didn't see his suggestion when you wrote your comment.

File: tests/api/utils/test_builder.py [hunk 1]
```diff
@@ -806,11 +806,11 @@ def test_kaniko_pod_spec_user_service_account_enrichment(monkeypatch):
 @pytest.mark.parametrize(
     "clone_target_dir,expected_workdir",
     [
-        (None, r"WORKDIR .*\/tmp.*\/mlrun"),
-        ("", r"WORKDIR .*\/tmp.*\/mlrun"),
-        ("./path/to/code", r"WORKDIR .*\/tmp.*\/mlrun\/path\/to\/code"),
-        ("rel_path", r"WORKDIR .*\/tmp.*\/mlrun\/rel_path"),
-        ("/some/workdir", r"WORKDIR \/some\/workdir"),
+        (None, r"WORKDIR /home/mlrun_code"),
```
Comment (alonmr) [inline tests/api/utils/test_builder.py:809]: there is no need to use regex anymore

**Review by alonmr — APPROVED** [review]
minor comment, other than that lgtm

**Pull request merged**